channeldb/payments.go

```diff
@@ -596,7 +596,8 @@ func (d *DB) QueryPayments(query PaymentsQuery) (PaymentsResponse, error) {
 		// Counting the total number of payments is expensive, since we
 		// literally have to traverse the cursor linearly, which can
 		// take quite a while. So it's an optional query parameter.
-		if query.CountTotal {
+		if query.CountTotal && query.CreationDateStart != 0 &&
```

Comment:
@coderabbitai generate unit tests for this if block

Reply:
Case 2 and 3 are covered by existing tests in `channeldb/payments_test.go`

Reply:
According to the original issue description, `CountTotal` should only be executed if we do not supply a creation start or end date. that means if we have `CreationDateStart` and `CreationDateEnd`, we want to count that transactions between these two dates, which is already done in the query handling block before we execute `CountTotal`. so `CountTotal` should not have the precedence if I understood the issue reporter's intent correctly